draft-ietf-privacypass-auth-scheme-extensions.md

```diff
@@ -123,17 +127,67 @@ issuance protocol. In particular, the value of an Extensions structure is provid
 metadata for the issuance protocol. Candidate issuance protocols are specified in
 {{?PUBLIC-ISSUANCE=I-D.hendrickson-privacypass-public-metadata}}.
 
-# Extensions Negotiation {#negotiation}
+# PrivateToken Challenge Extension Parameter
+
+As defined in {{Section 2.1 of AUTHSCHEME}}, the "PrivateToken" authentication
+scheme defines two parameters, "challenge" which contains the base64url-encoded
+TokenChallenge struct, and a "token-key" which contains the base64url-encoded
+public key used for this challenge. This document defines two OPTIONAL new parameters, 
+"extension-set," which contains the base64url-encoded representation of the
+following ExtensionSet structure, and "extensions" which contain the base64url-encoded 
+representation of the Extensions strucuture defined in {#extensions}. This document 
+follows the default padding behavior described in {{Section 3.2 of !RFC4648}}, so the
+base64url value MUST include padding.
+
+~~~
+struct {
+    enum { false(0), true(1) } Bool;
+    Bool is_required;
+    ExtensionType extension_type;
+} ExtensionEntry;
+
+struct {
+    ExtensionEntry extension_type;
+} ExtensionSet;
+
+enum {
+    reserved(0),
+    (65535)
+} ExtensionType;
+
+struct {
+    ExtensionType extension_types<0..2^16-1>;
+} ExtensionSet;
```

Comment:
Looks like this should be ExtensionEntry (and you should drop the one on 149:151?

Reply:
Alternatively:

struct {
    ExtensionType required_extensions<0..2^16-1>;
    ExtensionType optional_extensions<0..2^16-1>;
}

So that the client can just iterate through required_extensions to determine whether its even worth parsing the rest of the list?

Reply:
Addressed with the rename. I'd prefer to keep the ExtensionEntry struct, as I think this might adapt cleaner if we need fancier optionality later on. This preference isn't strong - feel free to push back.